core/solver/idr.cpp

```diff
@@ -170,6 +170,25 @@ void Idr<ValueType>::iterate(const LinOp *b, LinOp *x) const
 
     int total_iter = -1;
 
+    /* Memory movement summary for iteration with subspace dimension s
+     * Per iteration:
+     * (11/2s+10+5/(s+1))n * values + matrix/preconditioner storage
+     * For (s+1) iterations:
+     * (11/2s^2+31/2s+15)n * values + (s+1) * matrix/preconditioner storage
+     * dx SpMV:                    2(s+1)n * values + (s+1) * storage
+     * dx Preconditioner:          2(s+1)n * values + (s+1) * storage
+     * 1x multidot (gemm)           (s+1)n
+     * dx step 1 (fused axpys) s(s/2+5/2)n = approx 1 + sum k=[0,s) of (s-k+1)n
+     * dx step 2 (fused axpys) s(s/2+5/2)n = approx 1 + sum k=[0,s) of (s-k+1)n
+     * dx step 3:            s(9/2s+11/2)n = sum k=[0,s) of (8k+2+s-k+1+6)n
```

Comment:
After taking the theoretically optimum estimate for the GEMM, and using 0-based calculations, I get
```suggestion
     * dx step 3:            9/2s(s+1)n = sum k=[0,s) of (8k+2+s-k+6)n
```
Somehow the estimates of steps 1 and 2 don't change after switching to 0-based indexing, so they're still correct.

Reply:
Ah, never mind. It is indeed `s-k+1` over there. I made a silly error.